gui/index.go

```diff
@@ -44,27 +61,54 @@ func (ui *GUI) GetIndex(w http.ResponseWriter, r *http.Request) {
 
 	data.Address = address
 
-	minedWork, err := ui.hub.FetchMinedWork(true)
+	if len(address) != 35 {
+		data.Error = fmt.Sprintf("Address should be 35 characters")
+		ui.renderTemplate(w, r, "index", data)
+		return
+	}
+
+	accountID := *dividend.AccountID(address)
+
+	work, err := ui.hub.FetchMinedWorkByAddress(accountID)
 	if err != nil {
 		log.Error(err)
-		http.Error(w, "FetchMinedWork error: "+err.Error(),
+		http.Error(w, "FetchMinedWorkByAddress error: "+err.Error(),
 			http.StatusInternalServerError)
 		return
 	}
 
-	log.Tracef("mined work length is: %v", minedWork)
-
-	data.MinedWork = minedWork
-
-	resp, err := ui.hub.FetchAccountStats(address)
+	payments, err := ui.hub.FetchPaymentsForAddress(accountID)
 	if err != nil {
 		log.Error(err)
-		http.Error(w, "FetchAccountStats error: "+err.Error(),
+		http.Error(w, "FetchPaymentsForAddress error: "+err.Error(),
 			http.StatusInternalServerError)
 		return
 	}
 
-	data.AccountStats = resp
+	if len(work) == 0 && len(payments) == 0 {
+		log.Infof("Nothing found for address: %s", address)
+		data.Error = fmt.Sprintf("Nothing found for address: %s", address)
+		ui.renderTemplate(w, r, "index", data)
+		return
+	}
+
+	// // Reverse for display purposes. We want the most recent block/payments to be first.
+	reverseSlice(work)
```

Comment:
`ListMinedWork` already returns a reversed slice, it iterates and filters from the last entry to the first.